### ISSUE TITLE
The behavior of `zola serve` is not same as `zola build`

### ISSUE THREAD

**Issue by lewiszlw** [Keats/gutenberg#2120]
# Bug Report

## Environment

Zola version: 0.17.1

## Expected Behavior
Source files
```
content/games/tetris/index.md
content/games/tetris/tetris.js
```

When I exec `zola build`, the html file will be generated like below
```
public/games/tetris/index.html
```
So the template I created like below for importing js file correctly
```
...
<script type="module">
    import init from './{{ page.title }}.js'
    init()
</script>
...
```
When I open url `xxx/games/tetris`, the browser will request `games/tetris/tetris.js` as expected.

But when I exec `zola serve` locally and open url `xxx/games/tetris`, the browser will request `games/tetris.js` instead of `games/tetris/tetris.js`, the path is wrong and the request will get 404 Not Found.

## Step to reproduce
1. git clone https://github.com/NightsWatchGames/nightswatchgames.github.io.git
2. `zola serve` under the project
3. open `http://127.0.0.1:1111/games/tetris`

**Comment by Keats** [comment]
Ok so the issue is that the dev server doesn't redirect to the URL with a trailing `/` so the `./` path points to /game/tetris.js`.
Use `import init from '{{ page.path | safe }}{{page.title}}.js'` instead for a non relative URL

**Comment by lewiszlw** [comment]
<img width="1107" alt="image" src="https://user-images.githubusercontent.com/17514693/222024849-c4cee2a6-f31e-435b-99e1-c61e1a8000c2.png">

`games/tetris/tetris.js` will load `games/tetris/tetris_bg.wasm`, and `tetris_bg.wasm` will load assets with relative path.

<img width="762" alt="image" src="https://user-images.githubusercontent.com/17514693/222025322-106274eb-c25d-4ce9-8f5c-0638bd1f56a6.png">

`zola build` works as expected. (see https://nightswatchgames.github.io/games/tetris/).

**Comment by Keats** [comment]
Yes, it's because you're accessing the page as `/games/tetris` instead of `/games/tetris/` with the server. Both links load the page while in theory only the second should (I've made an issue to fix that, #2122 ).
When you're doing `import init from './{{ page.title }}.js'` in your template, it will not point to the same path depending on which URL you are using.
